@xen-orchestra/web-core/lib/components/ui/checkbox/UiCheckbox.vue

```diff
@@ -0,0 +1,429 @@
+<!-- v3 -->
+<template>
+  <!-- @TODO: Replace it by the VtsLabel component when update -->
+  <label class="form-checkbox" v-bind="wrapperAttrs">
+    <input
+      v-model="value"
+      :class="[classNames, { indeterminate: isIndeterminate }]"
+      :disabled="isDisabled"
+      type="checkbox"
+      class="ui-checkbox"
+      v-bind="attrs"
+    />
+    <span class="fake-checkbox">
+      <VtsIcon :fixed-width="false" :icon class="icon" accent="brand" />
+    </span>
+    <span v-if="slots.default" class="typo p1-regular">
+      <slot />
+    </span>
+  </label>
+  <VtsInfo v-if="slots.info" :accent>
+    <slot name="info" />
+  </VtsInfo>
+</template>
+
+<script lang="ts" setup>
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsInfo from '@core/components/info/VtsInfo.vue'
+import { useContext } from '@core/composables/context.composable'
+import { DisabledContext } from '@core/context'
+import { toVariants } from '@core/utils/to-variants.util'
+import { faCheck, faMinus } from '@fortawesome/free-solid-svg-icons'
+import { computed, type HTMLAttributes, useAttrs } from 'vue'
+
+type CheckboxAccent = 'brand' | 'success' | 'warning' | 'danger'
+
+defineOptions({ inheritAttrs: false })
+
+const props = withDefaults(
+  defineProps<{
+    accent: CheckboxAccent
+    disabled?: boolean
+    wrapperAttrs?: HTMLAttributes
+  }>(),
+  { disabled: undefined }
+)
+
+const value = defineModel<unknown>('modelValue')
+
+const slots = defineSlots<{
+  default?(): any
+  info?(): any
+}>()
+
+const isDisabled = useContext(DisabledContext, () => props.disabled)
+
+const classNames = computed(() => [
+  toVariants({
+    accent: props.accent,
+    disabled: isDisabled.value,
+  }),
+])
+const icon = computed(() => {
+  if (value.value === undefined) {
+    return faMinus
+  }
```

Comment:
We always want to display the `faMinus` if `isIndeterminate.value === true`.
IMHO, this avoids code duplication and improves code understanding
```suggestion
  if (isIndeterminate.value) {
    return faMinus
  }
```